Source/Core/PlaneGeometry.js

```diff
@@ -0,0 +1,380 @@
+
+        this._vertexFormat = vertexFormat;
+        this._workerName = 'createPlaneGeometry';
+    }
+
+    /**
+     * The number of elements used to pack the object into an array.
+     * @type {Number}
+     */
+    PlaneGeometry.packedLength = VertexFormat.packedLength;
+
+    /**
+     * Stores the provided instance into the provided array.
+     *
+     * @param {PlaneGeometry} value The value to pack.
+     * @param {Number[]} array The array to pack into.
+     * @param {Number} [startingIndex=0] The index into the array at which to start packing the elements.
+     *
+     * @returns {Number[]} The array that was packed into
+     */
+    PlaneGeometry.pack = function(value, array, startingIndex) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.object('value', value);
+        Check.defined('array', array);
+        //>>includeEnd('debug');
+
+        startingIndex = defaultValue(startingIndex, 0);
+
+        VertexFormat.pack(value._vertexFormat, array, startingIndex);
+
+        return array;
+    };
+
+    var scratchVertexFormat = new VertexFormat();
+    var scratchOptions = {
+        vertexFormat: scratchVertexFormat
+    };
+
+    /**
+     * Retrieves an instance from a packed array.
+     *
+     * @param {Number[]} array The packed array.
+     * @param {Number} [startingIndex=0] The starting index of the element to be unpacked.
+     * @param {PlaneGeometry} [result] The object into which to store the result.
+     * @returns {PlaneGeometry} The modified result parameter or a new PlaneGeometry instance if one was not provided.
+     */
+    PlaneGeometry.unpack = function(array, startingIndex, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('array', array);
+        //>>includeEnd('debug');
+
+        startingIndex = defaultValue(startingIndex, 0);
+
+        var vertexFormat = VertexFormat.unpack(array, startingIndex, scratchVertexFormat);
+
+        if (!defined(result)) {
+            return new PlaneGeometry(scratchOptions);
+        }
+
+        result._vertexFormat = VertexFormat.clone(vertexFormat, result._vertexFormat);
+
+        return result;
+    };
+
+    /**
+     * Computes the geometric representation of a plane, including its vertices, indices, and a bounding sphere.
+     *
+     * @param {PlaneGeometry} planeGeometry A description of the plane.
+     * @returns {Geometry|undefined} The computed vertices and indices.
+     */
+    PlaneGeometry.createGeometry = function(planeGeometry) {
+        var vertexFormat = planeGeometry._vertexFormat;
+
+        var min = new Cartesian3(-0.5, -0.5, 0.0);
+        var max = new Cartesian3( 0.5,  0.5, 0.0);
+
+        var attributes = new GeometryAttributes();
+        var indices;
+        var positions;
+
+        if (vertexFormat.position &&
+            (vertexFormat.st || vertexFormat.normal || vertexFormat.tangent || vertexFormat.bitangent)) {
+            if (vertexFormat.position) {
```

Comment:
Won't this always be true?  It's in the block for `if (vertexFormat.position && ...)